rxjs-1/leaderBoard.js

```diff
@@ -0,0 +1,65 @@
+const { fromEvent, from } = require('rxjs');
+const { reduce, map, throttleTime, flatMap, bufferCount } = require('rxjs/operators');
+const Table = require('easy-table');
+
+const { getRace } = require('./race');
+const { round } = require('./util');
+
+const race = getRace();
+const carNames = race.getCars();
+
+console.log('Participants:', carNames.join('/'));
+
+function sortByPosition(cars) {
+  return cars.sort((x, y) => y.xLocation - x.xLocation)
+}
+
+function raceReducer(acc, curr, index) {
+  const [leader, ...others] = acc
+  const isLast = index + 1 === carNames.length
+  return [
+    // reset first object once all gaps are calculated
+    ...(isLast ? [{ ...leader, leaderGapDistance: 0, leaderGapTime: 0 }, ...others] : acc),
+    {
+      position: index + 1,
+      carName: curr.carName,
+      leaderGapDistance: leader
+        ? round(leader.leaderGapDistance - curr.xLocation)
+        : curr.xLocation,
+      leaderGapTime: leader
+        ? round(leader.leaderGapTime - curr.time)
+        : curr.time,
+    },
+  ]
+}
+
+function getLeaderBoard(race) {
```

Comment:
Have you searched for a solution that would reuse `getCarSpeed`?

Reply:
Honestly I haven't cos I though it was two different exercices... 
Do you want me to find to another solution ? The current structure has to be a bit modified.

Reply:
no no, it was just a question around your thought process more than an actual thing that I wan't you to code 🙂